File: projects/10/ruby/Walking.rb
#!/usr/bin/ruby
# Walking.rb defines a walking superclass to be extended by specific bird sub-classes
# Begun by: Dr. Adams, for CS 214 at Calvin College
# Completed by: Ian Christensen, for CS 214 at Calvin College
# Date: Spring, 2018

require './Bird.rb'

class Walking < Bird
	
	# Method: move, gets the movement of the bird
	# Precondition: Bird must exist
	# Receives: a Bird object
	# Returns: 'walked passed', the walking movement
	def move
		'walked passed'
	end
end

File: projects/10/java/Walking.java
/* Walking.java provides a Walking subclass of Bird.
 *
 * Begun by: Prof. Adams, for CS 214 at Calvin College.
 * Completed by: Ian Christensen, for CS 214 at Calvin College
 * Date: Spring, 2018
 */

public class Walking extends Bird {
    
	/* explicit constructor
	 * Receive: name, a String 
	 * PostCond: myName == name.
	 */
	public Walking() {
		super();
	}

	public Walking(String name) {     
		super(name);
	} 

	/* A Walking's move
	 * Return: a Walking-move ("walked passed").
	 */
	public String move() {
		return "walked passed";
	} 
}
